src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -69,15 +76,33 @@ internal sealed class VisualStudioProject
         /// </summary>
         private readonly FileChangeWatcher.IContext _fileReferenceChangeContext;
 
+        /// <summary>
+        /// track whether we have been subscribed to <see cref="IDynamicFileInfoProvider.Updated"/> event
+        /// </summary>
+        private readonly HashSet<IDynamicFileInfoProvider> _eventSubscriptionTracker = new HashSet<IDynamicFileInfoProvider>();
```

Comment:
lower down you access this in a lock, but tehre is no information at this level as to the threading contract around this variable and the members that access it.

Reply:
following pattern of all other fields here.